Library/ViewModels/DiscoveryViewModel.swift

```diff
@@ -170,6 +180,12 @@ DiscoveryViewModelOutputs {
   public func viewDidLoad() {
     self.viewDidLoadProperty.value = ()
   }
+
+  private let navigationHeaderTopInsetProperty = MutableProperty<CGFloat?>(nil)
+  public func viewSafeAreaInsetsDidChange() {
+    self.navigationHeaderTopInsetProperty.value = 20
```

Comment:
Not sure if we want to use a constant like this - what is the `20`, the status bar height?

Reply:
Yeah, I tried using the statusBar.frame.height, but the problem is that the statusBar height is different on iPhone X.

Reply:
I think what you want to do is change your input to be:

```swift
  private let viewSafeAreaInsetsDidChangeProperty = MutableProperty<UIEdgeInsets?>(nil)
  public func viewSafeAreaInsetsDidChange(to insets: UIEdgeInsets) {
    self.viewSafeAreaInsetsDidChangeProperty.value = insets
  }
```

And then your output:

```swift
self.navigationHeaderTopInset = self.viewSafeAreaInsetsDidChangeProperty.signal.skipNil()
      .map { $0.top }
      .skipRepeats()
```

As far as I can tell that should give the same desired result and not base it on a constant value.

Reply:
Actually the `insets.top` will return the same value as the statusBar. So it would have the same result as if we used the statusBar height directly